OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -2064,18 +2070,38 @@ public static void handleNotificationOpen(Context inContext, JSONArray data, boo
 
       runNotificationOpenedCallback(data, true, fromAlert);
 
-      // Open/Resume app when opening the notification.
-      if (!fromAlert && !urlOpened && !defaultOpenActionDisabled)
-         startOrResumeApp(inContext);
+      // Check if the notification click should lead to a DIRECT session
+      if (sessionManager != null && shouldInitDirectSessionFromNotificationOpen(inContext, fromAlert, urlOpened, defaultOpenActionDisabled)) {
+         // We want to set the app entry state to NOTIFICATION_CLICK when coming from background
+         appEntryState = AppEntryAction.NOTIFICATION_CLICK;
+         sessionManager.onDirectSessionFromNotificationOpen(notificationId);
```

Comment:
i think this check if the application was on the foreground right? so in theory onDirectSessionFromNotificationOpen wont be set if the app was on foreground and the notification was clicked

Reply:
But if `sessionManager` was null it would crash anyway. Maybe we should check if it is null and then initialize it if it is?

Reply:
that can be an option

Reply:
We don't want a notification click when the app is in the foreground to change the session, only background clicks